### PR TITLE
Add switch-mirgecom script by @inducer

### PR DESCRIPTION
**Pull request by matthiasdiener**

### CODE REVIEW

File: switch-mirgecom.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    echo "-------------------------------------------------------------------"
+    echo "Updating $pkg to $branch..."
+    echo "-------------------------------------------------------------------"
+
+    (
+    cd "$pkg"
+    git fetch "$remote"
+
+    git checkout "$branch"
```
Comment (inducer) [inline switch-mirgecom.sh:54]: You'll need something to create the remote, probably. I just rely on remote names that exist for me but that will likely not exist for others.

File: switch-mirgecom.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    remote_and_branch="${PACKAGE_TO_REMOTE_AND_BRANCH[$pkg]}"
+    remote="$(dirname "$remote_and_branch")"
+    branch="$(basename "$remote_and_branch")"
+
+    echo "-------------------------------------------------------------------"
+    echo "Updating $pkg to $branch..."
+    echo "-------------------------------------------------------------------"
+
+    (
+    cd "$pkg"
```
Comment (inducer) [inline switch-mirgecom.sh:51]: Should probably check that you're running in the expected directory.

File: switch-mirgecom.sh (hunk 1 above)
Comment (matthiasdiener) [reply switch-mirgecom.sh:57]: Should be fixed in bae9609472542d0210e50b39765f127ca5738adb

File: switch-mirgecom.sh (hunk 2 above)
Comment (matthiasdiener) [reply switch-mirgecom.sh]: Not sure what you mean - check that we are running the script from the emirge directory? The `cd` here would just fail.

File: switch-mirgecom.sh [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+if [[ $# -ne 1 ]]; then
+    echo "Usage: $0 <env_name>"
+    echo 'Where <env_name> is one of "parlazy", "prod", "main".'
+    exit 1
+fi
+
+envname="$1"
+if [[ "$envname" = "parlazy" ]]; then
+    declare -A PACKAGE_TO_REMOTE_AND_BRANCH=( \
```
[the remark this answers is not in the document]
Comment (matthiasdiener) [reply switch-mirgecom.sh]: fixed in bae9609472542d0210e50b39765f127ca5738adb

File: switch-mirgecom.sh (hunk 2 above)
Comment (inducer) [reply switch-mirgecom.sh:53]: Fair enough.

**Review by inducer — APPROVED** [review]
Haven't run it, but LGTM.

**Pull request merged**